Tools/TestWebKitAPI/Tests/WebKitCocoa/RunOpenPanel.mm

```diff
@@ -79,6 +93,43 @@ - (void)webView:(WKWebView *)webView runOpenPanelWithParameters:(WKOpenPanelPara
     }];
     Util::run(&testFinished);
 }
+
+TEST(WebKit, FileInputTypeCancelEvent)
+{
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:NSMakeRect(0, 0, 100, 100)]);
+    auto uiDelegate = adoptNS([[FileInputTypeCancelEventUIDelegate alloc] init]);
+    [webView setUIDelegate:uiDelegate.get()];
+
+    NSString *markup = @""
+        "<script>"
+        "    function loaded() {"
+        "        setTimeout(() => {"
+        "            const $file = document.getElementById('file');"
+        "            $file.addEventListener('cancel', () => {"
+        "                webkit.messageHandlers.testHandler.postMessage('cancel');"
+        "            }, false);"
+        "        }, 0);"
+        "    }"
+        "</script>"
+        "<body style='width: 100vw; height: 100vh;' onload='loaded()'>"
+        "   <input style='width: 100vw; height: 100vh;' type='file' id='file' name='file'>"
+        "</body>";
+
+    [webView loadHTMLString:markup baseURL:nil];
+    [webView _test_waitForDidFinishNavigation];
+
+    __block bool done = false;
+    [webView performAfterReceivingMessage:@"cancel" action:^{
+        done = true;
+    }];
+
+    NSPoint clickPoint = NSMakePoint(50, 50);
+    NSInteger windowNumber = [webView window].windowNumber;
+    [webView mouseDown:[NSEvent mouseEventWithType:NSEventTypeLeftMouseDown location:clickPoint modifierFlags:0 timestamp:0 windowNumber:windowNumber context:nil eventNumber:0 clickCount:1 pressure:1]];
+    [webView mouseUp:[NSEvent mouseEventWithType:NSEventTypeLeftMouseUp location:clickPoint modifierFlags:0 timestamp:0 windowNumber:windowNumber context:nil eventNumber:0 clickCount:1 pressure:1]];
```

Comment:
`sendClickAtPoint`

Reply:
@pxlcoder what does this comment mean?

Reply:
There's a `sendClickAtPoint` method which would simplify this logic.